Makefile

```diff
@@ -70,7 +70,7 @@ golay: golay.o parity.o
 	$(CC) $(LDFLAGS) -o $@ $^
 
 unoptar: unoptar.o common.o golay_codes.o parity.o
-	$(CC) -o $@ -L/usr/local/lib $^ -lm -lpng -lz
+	$(CC) -o $@ -L/usr/local/lib -L/opt/homebrew/lib $^ -lm -lpng -lz
```

Comment:
I'm concerned that this won't work correctly on all platforms, but it definitely does need to change with the separation of `/usr/local` for macOS on Intel and `/opt/homebrew` for macOS on ARM.

The solution is likely to dynamically detect the presence of Homebrew and use it entirely instead:

```Makefile
DEFAULT_LIB_DIR=/usr/local/lib

ifeq ($(shell which brew),)
# brew isn't installed, use the default
LIB_DIR ?= $(DEFAULT_LIB_DIR)
else
# brew is installed
LIB_DIR ?= $(shell brew --prefix)/lib
endif

unoptar: unoptar.o common.o golay_codes.o parity.o
	$(CC) -o $@ -L$(LIB_DIR) $^ -lm -lpng -lz

```
Do you have time to do up something like that?

Reply:
Yes, however supplying severel -L or -I paths is perfectly fine, and even doesn't disturb if the given paths are not existing AFAIK, so that patch is perfect as it is IMHO.